lib/ansible/module_utils/urls.py

```diff
@@ -302,6 +302,16 @@ def match_hostname(cert, hostname):
 
     HAS_MATCH_HOSTNAME = True
 
+if hasattr(socket, 'create_connection'):
+    create_connection = socket.create_connection
+else:
+    # Just a graceful fallback for py < 2.6, implementing old behaviour.
+    # Wouldn't work with IPv6, but it is unlikely that python 2.4 could be met
+    # on IPv6-only machine.
+    def create_connection(address):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(address)
+        return sock
```

Comment:
I'm going to leave this in needs_revision.  My requirements for considering this change have been outlined above.